### PR TITLE
Added confirmation messages for Reject Game and Cancel Game links

### LINKED ISSUE #1767: When you cancel a game, include a link to the game, in case you mis-clicked
When you create a game, and then cancel it, the "Successfully rejected game" text at the top" should instead say "Successfully canceled Game N", with a link, so that if you clicked cancel by mistake, you could click the link, get the game page, and click something like "create a new game against with the same parameters", much like the "create a new game" follow-up buttons after a completed game. So this would depend on the rejected-game page having some more stuff on it, but once it did, insta-dismissing but linking to that seems good.

### LINKED ISSUE #1777: Prompt for confirmation before rejecting a game
Re-creating a rejected game is rather annoying now, and will probably at least somewhat annoying in the future, so it'd be nice if it prompted for confirmation, like surrendering.

### PR DESCRIPTION
**Pull request by blackshadowshade**
Fixes #1777. Fixes #1767.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/989/

### CODE REVIEW

**Comment by cgolubi1** [comment]
Sure, this isn't complicated, and thanks for garbage-collecting the Overview.js functions.

I'll bring it up in virtualbox and make sure it works, then merge it.

**Comment by cgolubi1** [comment]
So, i don't wanna be picky, *but* i think this is probably not okay:
cancel_cancel_cancel

The user has just clicked a button which says "Cancel", and we're prompting them "Are you sure you want to cancel the game?" and we're showing them two buttons, "OK" which cancels the game, and "Cancel" which cancels the action to cancel the game - we can't use the same word to mean two opposite things in the same dialogue.

**Comment by cgolubi1** [comment]
Not to beat on my hobbyhorse from 10 minutes ago, but i wonder if it's too late to get rid of the word "Cancel".  You could easily imagine this link saying "Reject your own game" (it's not like we don't have the horizontal space, now that this action no longer needs to be taken from the overview page), or something more terse that someone else comes up with, but i'll throw "Reject your own game" out there as a suggestion, or even "Reject the game you proposed"

Basically, my thinking is that getting rid of the word "Cancel" as a verb that someone can do to a new game will make us happier in the long-term, and in the short term, it would solve this particular problem.

**Comment by cgolubi1** [comment]
Anyway, the functionality works:
* Cancelling a game succeeds
* Choosing not to cancel a game succeeds
* Rejecting a game succeeds
* Clicking Reject, clicking Cancel in the dialogue, and then accepting the game succeeds

So that's all good, let's just talk about the word Cancel.

**Comment by irilyth** [comment]
Heh: I was just thinking that we should consolidate down to one term, and that "cancel" was much better than "reject ". I see your point about the Ok vs Cancel thing, though. Hmm.

-Josh (from my phone)

**Comment by AdmiralJota** [comment]
Retract, Rescind, Revoke, Call Off, Withdraw...

**Comment by irilyth** [comment]
One thing I like about "cancel" is that it feels like it applies equally well to the creator and the opponent. Things like "retract" and "revoke" doesn't really sound to me like what I'm doing if you propose a game and I don't accept it. (And vice versa "reject".)

**Comment by blackshadowshade** [comment]
Okay, we're going with Reject/Withdraw for the active verbs, and Cancelled for past participle.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/992/ (if it passes)

**Comment by cgolubi1** [comment]
Okay, great, let me:
* bring this up in virtualbox
* see if i can find the word "Cancel" in the UI to kick around any more

**Comment by cgolubi1** [comment]
The front UI page still says "REJECTED".  Is that what we want?
front_ui_says_rejected

**Comment by cgolubi1** [comment]
Looks fine otherwise.  My vote is that changing to CANCELLED on the front page would be more consistent with "two verbs and a noun".  It's also more correct in the case in which the proposing player withdraws their own game and the target player sees that game in the closed column, since that game was technically withdrawn not rejected.

**Comment by cgolubi1** [comment]
Adding "dev-blocker" to remind myself that we should fully standardize on Withdraw/Reject/Cancelled before pushing to dev, because testers will be confused by an intermediate state.

**Comment by irilyth** [comment]
> My vote is that changing to CANCELLED on the front page would be more consistent with "two verbs and a noun".

Sounds good to me, except that it turns out "cancelled" is actually an adjective, not a noun.

**Comment by cgolubi1** [comment]
> Sounds good to me, except that it turns out "cancelled" is actually an adjective, not a noun.

I reject your narrow concepts of grammar.

**Comment by blackshadowshade** [comment]
Okay, try this.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/995/ (if it passes)

**Comment by cgolubi1** [comment]
Thanks.  Let's try this out.
